index.js

```diff
@@ -32,12 +53,59 @@ app.get('/channels',(req,res)=>{
     else{
         mirth_api.channels()
         .then(stats=>{
-            res.render('channels',{channels : stats.channels, channelGroups : stats.channelGroups})
+            //copy fetched stats
+            fetched_stats = [...stats.channels]
+
+            //delete existing stats from table
+            // console.log('deleting existing stats.')
+            // DB_CONNECTION.run(`DELETE FROM stats`,[],err => {
+            //     if(err) return console.log(err.message)
+            // })
+
+            //insert new stats into table
+            // stats.channels.forEach(channel => {
+            //     sql = `INSERT INTO stats(channelId, channelName, received, sent, error, filtered) VALUES (?,?,?,?,?,?)`
+            //     DB_CONNECTION.run(sql, [channel.channelId, channel.channelName, channel.received, channel.sent, channel.error, channel.filtered], err => {
+            //         if(err) return console.log(err.message)
+            //     })
+            // })
+            //newly stored stats
+            // console.log('newly stored stats:')
+            // DB_CONNECTION.all(`SELECT * FROM stats`,[],(err, rows)=>{
+            //     if(err) return console.log(err.message)
+            //     rows.forEach(row => {
+            //         console.log(row)
+            //     })
+            // })
+            res.render('channels',{channels : fetched_stats, channelGroups : stats.channelGroups})
         })
         .catch(error => res.send("Cannot fetch channels!! Service temporarily unavailable."))
     }
 })
 
+app.get('/save',(req,res)=>{
+    if(token==''){
+        res.render('index', {msg : null})
+    }
+    else{
+        console.log('writing to database')
+        try{
+            let timestamp = Date.now()
+            console.log('timestamp now: '+ timestamp)
+            fetched_stats.forEach(channel => {
+                sql = `INSERT INTO stats(timestamp,channelId, channelName, received, sent, error, filtered) VALUES (?,?,?,?,?,?,?)`
```

Comment:
create a separate file to write queries. call those functions from here.
you can name that file model.js